pyblish_lite/control.py

```diff
@@ -58,40 +66,107 @@ def __init__(self, parent=None):
             "ordersWithError": set()
         }
 
-    def reset(self):
-        """Discover plug-ins and run collection"""
+    def prepare_for_reset(self):
+        self.validated = False
+        self.extracted = False
+        self.publishing = False
+
         self.context = pyblish.api.Context()
-        self.plugins = pyblish.api.discover()
+        self.context.data["optional"] = False
+        self.context.data["publish"] = True
 
-        self.was_discovered.emit()
+        self.context.data["label"] = 'Context'
+        self.context.data["name"] = 'Context'
+        self.context.data["_type"] = 'context'
 
-        self.pair_generator = None
-        self.current_pair = (None, None)
+        self.context.data["icon"] = 'book'
+
+        self.context.data["_has_succeeded"] = False
```

Comment:
Are you storing delegate state inside of the context object? Does the plug-ins use this context too? If so, it can interfere with user-created data. Don't store GUI-related data inside of the Pyblish data structures.

Reply:
Well that icon can be changed - e.g. with your collector above...

Reply:
I think the question is not about the icon, but the `_has_suceeded` state

Reply:
Yes, the state. Better store this in the instance of the class itself.